### PR TITLE
API compare_and_swap

### PR DESCRIPTION
**Pull request by drystone**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -750,6 +756,38 @@ impl Gdbm {
...
+    // API: compare_and_swap
+    pub fn compare_and_swap(
+        &mut self,
+        key: Vec<u8>,
+        old: Option<&[u8]>,
+        new: Option<Vec<u8>>,
+    ) -> io::Result<CompareAndSwapSummary> {
+        let current = self.get(&key)?;
+
+        if old != current.as_deref() {
```
Comment (jgarzik) [inline src/lib.rs:769]: Help me understand this specific line of code.  ELI5.  :)

I would understand "if old ==" but I do not understand "if old !="

Comment (drystone) [reply src/lib.rs:769]: We only swap if `old` (our parameter) isn't the same as `current` (the value in the database). That's the compare bit of compare and swap, at least as far as I understood from [sled's documentation](https://docs.rs/sled/latest/sled/struct.Db.html#method.compare_and_swap).

Comment (drystone) [reply src/lib.rs:769]: Haha @jgarzik , hopefully you've seen sense and decided not to merge - this pattern sucks :)

**Comment by jgarzik** [comment]
We'll leave this out for now.  Thanks for exploring.
